oauth-common/src/main/java/io/strimzi/kafka/oauth/common/JSONUtil.java

```diff
@@ -19,23 +22,64 @@
         return MAPPER.readValue(is, clazz);
     }
 
-    public static String getClaimFromJWT(String claim, Object token) {
+    /**
+     * Convert object to JsonNode
+     *
+     * @param value Json-serializable object
+     * @return Object as JsonNode
+     */
+    public static JsonNode asJson(Object value) {
+        if (value instanceof JsonNode)
+            return (JsonNode) value;
+
+        // We re-serialise and deserialize into generic json object
         try {
-            // No nice way to get arbitrary claim from already parsed token
-            // therefore we re-serialise and deserialize into generic json object
-            String jsonString = JsonSerialization.writeValueAsString(token);
-            JsonNode node = JsonSerialization.readValue(jsonString, JsonNode.class);
-            JsonNode claimNode = node.get(claim);
-
-            if (claimNode == null) {
-                throw new RuntimeException("Access token contains no '" + claim + "' claim: " + jsonString);
-            }
+            String jsonString = JsonSerialization.writeValueAsString(value);
+            return JsonSerialization.readValue(jsonString, JsonNode.class);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to convert value to JSON (" + value + ")", e);
+        }
+    }
 
-            return claimNode.asText();
+    /**
+     * Get specific claim from token.
+     *
+     * @param claim jq style query where nested names are specified using '.' as separator
+     * @param token parsed object
+     * @return Value of the specific claim as String or null if claim not present
+     */
+    public static String getClaimFromJWT(String claim, Object token) {
+        // No nice way to get arbitrary claim from already parsed token
+        JsonNode node = asJson(token);
+        return getClaimFromJWT(node, claim.split("\\."));
+    }
 
-        } catch (IOException e) {
-            throw new RuntimeException("Failed to read '" + claim + "' claim from token", e);
+    /**
+     * Get specific claim from token.
+     *
+     * @param node parsed JWT token payload
+     * @param path name segments where all but last should each point to the next nested object
+     * @return Value of the specific claim as String or null if claim not present
+     */
+    public static String getClaimFromJWT(JsonNode node, String... path) {
+        for (String p: path) {
+            node = node.get(p);
+            if (node == null) {
+                return null;
+            }
         }
+        return node.asText();
     }
 
+    public static List<String> asListOfString(JsonNode arrayNode) {
+        if (!arrayNode.isArray()) {
+            throw new IllegalArgumentException("JsonNode not an array node: " + arrayNode);
+        }
+        ArrayList<String> result = new ArrayList<>();
+        Iterator<JsonNode> it = arrayNode.iterator();
+        while (it.hasNext()) {
+            result.add(it.next().asText());
```

Comment:
IAE for non-array, but not for non-String member of the array?

Reply:
We convert elements to strings and they get validated in the calling `authorize()` method immediately afterwards. Helper method itself is potentially more universally usable - for some other occasion maybe, if it doesn't enforce by itself that elements have to be strings.

Reply:
I improved this a little bit as it turns out _asText()_ does not just automatically convert object node to json string.